### PR TITLE
Accept C-PgUp, C-PgDown for moving between tabs

### PR DESCRIPTION
**Pull request by jmc-88**
It bugs me that I can't cycle between tabs with C-PgUp and C-PgDown, so here's a fix.

### CODE REVIEW

**Comment by jeremypw** [comment]
It is recommended you make a new branch from your fork of master and make you changes there.  You can then keep your master branch in sync with elementary:master and make other branches for other fixes.

**Comment by jeremypw** [comment]
This conflicts with the existing behaviour for Ctrl-PageUp and Ctrl-PageDown, which is to move to the beginning/end of the current line.

File: src/MainWindow.vala [hunk 1]
```diff
@@ -981,6 +989,18 @@ namespace Scratch {
 
             project_pane.visible = !project_pane.visible;
         }
+
+        private void action_focus_next_document () {
+            Scratch.Widgets.DocumentView? view =
```
Comment (jeremypw) [inline src/MainWindow.vala]: You can just use `var view = ` here as the type is indicated by the rhs.

File: src/MainWindow.vala [hunk 2]
```diff
@@ -981,6 +989,18 @@ namespace Scratch {
...
+                split_view.get_focus_child () as Scratch.Widgets.DocumentView;
+            view.next_document ();
```
Comment (jeremypw) [inline src/MainWindow.vala]: You have not dealt with the possibility of a null here.

File: src/MainWindow.vala [hunk 3]
```diff
@@ -981,6 +989,18 @@ namespace Scratch {
...
+        }
+
+        private void action_focus_previous_document () {
+            Scratch.Widgets.DocumentView? view =
+                split_view.get_focus_child () as Scratch.Widgets.DocumentView;
+            view.previous_document ();
+        }
```
Comment (jeremypw) [inline src/MainWindow.vala]: See comments above for related function.

**Comment by jeremypw** [comment]
It is best to raise this kind of thing as an issue before coding to see if there is support/design team approval.

**Comment by jmc-88** [comment]
> This conflicts with the existing behaviour for `<Ctrl><PageUp>` and , which is to move to the beginning/end of the current line. Code uses `<Ctrl><Alt><PageUp>` and `<Ctrl><Alt><PageUp>` to switch tabs (like gedit).

I would've swapped the two, but good to know it's already supported. Given this, I suppose this change is not needed. Thanks anyway for having a look at it.

**Comment by jeremypw** [comment]
Hmm, having said that, I cannot get <Control><Alt><PageUp>/<Control><Alt><PageDown> to work now :-( - not sure what changed.  So there is work to be done to make behaviour of Code, Terminal, Files etc consistent in this area.

**Pull request closed without merging**